_posts/2018-03-31-rereading-the-mythical-man-month.md

```diff
@@ -0,0 +1,207 @@
+
+Now I've had the same experience with <a href="http://wiki.c2.com/?MythicalManMonth" target="_blank">The Mythical Man-Month</a> by <a href="http://wiki.c2.com/?FredBrooks" target="_blank">Fred Brooks</a>, the father of the IBM System/360. This book was published in **1975** and it was reissued in its 20th Anniversary with four new chapters. On the one hand, it contains problems which still happen nowadays and it's worthwhile reading, because:
+
+> Those who cannot remember the past are condemned to repeat it. -- Jorge Santayana
+
+On the other hand, it mentions concepts that I didn't realize for the first time I read it when I was studying in 1998. Here you'll find some of my notes about it.
+
+# Software as a craft
+
+The book starts talking about **the craft of system programming** and the joys and woes inherent in it. Among joys, we find: **always learning**. 
+
+When explaining the problems with the size of programs: _That requires invention and craftsmanship. (...) So the first area of craftsmanship is in trading function for size. (...) The second area of craftsmanship is space-time trade-offs._
+
+When talking about the need of paperwork in project management: _To the new manager, fresh from operating as a craftsman himself, these seem an unmitigated nuisance, (...)_.
+
+Let's remember this book was written in **1975** and terms like _craftsmanship_ and _craftsman_ were used.
+
+# Brooks's Law
+
+Software projects suffered delays because of these reasons:
+
+* **Optimism**
+
+_'All will go well'_, _'This time it will surely run'_, _'I just found the last bug'_, those thoughts sound familiar, don't they? I love the following sentence, because it's related to evolutionary designs:  _The incompletenesses and inconsistencies of our ideas become clear only during implementation._ 
+
+* **The Man Month**
+
+This was the unit of effort used in estimating and scheduling. Men and months were interchanged, when confusing effort with progress. It's said that they can be only interchanged when a task can be partitioned among many workers with no communications among them and no sequential constraints.
+
+* **Systems Test**
+
+Testing phase was the most mis-scheduled part of programming, because it was at the end of the development process. It was thought the solution was to increase the system test time in the original schedule. 
+
+* **Gutless Estimating**
+
+When an omelette, promised in two minutes, has not set in two minutes, the customer has two choices: wait or eat it raw (or partially burned, when turning up the heat). Software managers might have the same courage and firmness than chefs, avoiding false scheduling. _We need to develop and publicize productivity figures, bug-incidence figures, estimating rules, and so on. The whole profession can only profit from sharing such data._
```

Comment:
Software managers might have the same courage and firmness than chefs -> Software managers might have the same courage and firmness as do chefs